packages/app-builder-lib/src/electron/electronMac.ts

```diff
@@ -223,6 +223,9 @@ export async function createMacApp(packager: MacPackager, appOutDir: string, asa
       }
       return result
     })
+
+    // `CFBundleDocumentTypes` may be defined in `mac.extendInfo`, so we need to merge it in that case
+    appPlist.CFBundleDocumentTypes = [...(appPlist.CFBundleDocumentTypes || []), ...documentTypes]
```

Comment:
Probably should also handle duplicate entries.
```
Array.from(new Set([...(appPlist.CFBundleDocumentTypes || []), ...documentTypes]))
```
Not sure if there's a more pretty way to write that, can't recall the `Set` constructor off the top of my head

Reply:
Thanks @mmaietta! 🙏

The items in the `CFBundleDocumentTypes` array are objects (not primitives) so I don't think this will work. The objects will not be recognised as equal.

```js
Array.from(new Set([{ foo: 'bar'}, { foo: 'bar'}]))
// -> [{ "foo": "bar" }, { "foo": "bar" }]
```

I can create a custom function to do de-duplication if you like. I think it's probably overkill in this case but happy to do it if you would like me to?